### ISSUE TITLE
Unintuitive behaviour of options taking a parameter when a default value is given

### ISSUE THREAD

**Issue by srvsh** [celogeek/MooX-Options#49]
In the following code, the `name` option takes a string parameter and also has a default value 'UNKNOWN':

``` perl
# test.pl
use strict;
use warnings;

package MyApp;

use Moo;
use MooX::Cmd;
use MooX::Options;

option name => (
    is      => 'ro',
    format  => 's',
    default => 'UNKNOWN',
);

sub execute {
    my ($self, $args) = @_;

    print "Name: " . $self->name . "\n";
}

1;

package main;

MyApp->new_with_cmd;
```

When the code is run as `perl test.pl --name` (no name passed), the script runs successfully and sets `name` to the default value. The output is:

```
Name: UNKNOWN
```

Ideally I'd expect an error to be thrown since `--name` is not provided with a parameter.

**Comment by celogeek** [comment]
In fact you pass an "undef" params.

Can you try with "required => 1" ?

**Comment by srvsh** [comment]
'UNKNOWN' instead of `undef` was just for an example :)

There's no change in behaviour with `required => 1`. If I don't pass the `default` value, then the it correctly shows a error message saying `name is missing`.

**Comment by celogeek** [comment]
ok i need to check this.

**Comment by srvsh** [comment]
@celogeek I have fixed this issue and one more in PR #50 . Please take a look :)

**Comment by rehsack** [comment]
Seems to be resolved
